lib/gooddata/lcm/actions/associate_clients.rb

```diff
@@ -46,7 +49,35 @@ class AssociateClients < BaseAction
       ]
 
       class << self
+        DELETE_MODES = %w(
+          none
+          delete_projects
+          delete_extra
+        )
         def call(params)
+
+          unless DELETE_MODES.include?(params.delete_mode) || params.delete_mode.nil?
```

Comment:
Refers to 'params' more than self (maybe move it to another class?) (FeatureEnvy)